packages/yarnpkg-fslib/sources/index.ts

```diff
@@ -288,32 +288,28 @@ export type XFS = NodeFS & {
 
   mktempPromise(): Promise<PortablePath>;
   mktempPromise<T>(cb: (p: PortablePath) => Promise<T>): Promise<T>;
+
+  /**
+   * Unregisters and tries to remove all temp folders created by mktempSync and mktempPromise
+   */
+  rmtempPromise(): Promise<void>;
+
+  /**
+   * Unregisters and tries to remove all temp folders created by mktempSync and mktempPromise
+   */
+  rmtempSync(): void;
 };
 
 const tmpdirs = new Set<PortablePath>();
 
 let cleanExitRegistered = false;
 
 function registerCleanExit() {
-  if (!cleanExitRegistered)
-    cleanExitRegistered = true;
-  else
+  if (cleanExitRegistered)
     return;
 
-  const cleanExit = () => {
-    process.off(`exit`, cleanExit);
-
-    for (const p of tmpdirs) {
-      tmpdirs.delete(p);
-      try {
-        xfs.removeSync(p);
-      } catch {
-        // Too bad if there's an error
-      }
-    }
-  };
-
-  process.on(`exit`, cleanExit);
+  cleanExitRegistered = true;
+  process.once(`exit`, xfs.rmtempSync);
```

Comment:
As a rule I prefer to always use explicit callbacks regardless of whether the arguments happen to be mostly compatible or not, to avoid the kind of problems you get with `.map(something)`.
```suggestion
  process.once(`exit`, () => {
    xfs.rmtempSync();
  });
```